File: packages/shared-types/src/club.ts
export type ClubMemberRole = "coordenador" | "preparador" | "analista";

export interface Club {
  id: string;
  name: string;
  official_email_domain: string;
  verified_at: string | null;
  created_at: string;
}

export interface ClubMember {
  id: string;
  club_id: string;
  user_id: string;
  role: ClubMemberRole;
  created_at: string;
}

export interface Team {
  id: string;
  club_id: string;
  name: string;
  season: string | null;
  created_at: string;
}

export interface Athlete {
  id: string;
  club_id: string | null;
  team_id: string | null;
  user_id: string | null;
  full_name: string;
  birth_date: string | null;
  position: string | null;
  jersey_number: number | null;
  created_at: string;
}

File: apps/api/src/schemas/club.py
from datetime import datetime

from pydantic import BaseModel, EmailStr


class ClubCreate(BaseModel):
    name: str
    official_email_domain: str
    admin_full_name: str
    admin_email: EmailStr


class ClubOut(BaseModel):
    id: str
    name: str
    official_email_domain: str
    verified_at: datetime | None
    created_at: datetime


class ClubMemberInvite(BaseModel):
    email: EmailStr
    role: str  # coordenador | preparador | analista


class ClubMemberOut(BaseModel):
    id: str
    club_id: str
    user_id: str
    role: str
    created_at: datetime
